frontends/p4/toP4/toP4.cpp

```diff
@@ -688,6 +689,12 @@ bool ToP4::preorder(const IR::Declaration_Variable *v) {
 
 bool ToP4::preorder(const IR::Type_Error *d) {
     dump(1);
+
+    if (!isDeclaration) {
+        builder.append("error");
```

Comment:
I think we just need always emit `error` regardless of `isDeclaration`. And if `isDeclaration` is true print nested declarations. This would eliminate illogical `builder.append("error ")` inside the loop

Reply:
`builder.append("error ")` inside the loop is still needed to conditionally emit error declarations outside of core.p4. Always emitting the `error` might produce invalid program with `error {}`.

Reply:
But you can refactor the code, so it would work,no?